mono/metadata/threadpool.c

```diff
@@ -332,9 +331,12 @@ worker_callback (gpointer unused)
 	while (!mono_runtime_is_shutting_down ()) {
 		gboolean retire = FALSE;
 
-		if ((thread->state & (ThreadState_AbortRequested | ThreadState_SuspendRequested)) != 0) {
+		if (mono_thread_interruption_requested ()) {
```

Comment:
No, it's not on purpose. Is there a function that would let me check if there is any interruption of any kind (Suspend, Abort, Stop).
I can't just call `mono_thread_interruption_checkpoint` because that mean I would have to unlock and lock back `domains_lock` and that is pretty wasteful, and I can't just call it inside the lock, as that could deadlock in case of Suspend.